worlds/metroidprime/Regions.py

```diff
@@ -0,0 +1,102 @@
+import typing
+
+from .Locations import MetroidPrimeLocation
+
+from .data.RoomNames import RoomName
+from BaseClasses import CollectionState, Region
+
+if typing.TYPE_CHECKING:
+    from . import MetroidPrimeWorld
+
+
+def create_regions(world: "MetroidPrimeWorld", final_boss_selection: int):
+    # create all regions and populate with locations
+    menu = Region("Menu", world.player, world.multiworld)
+    world.multiworld.regions.append(menu)
+
+    for area_data in world.game_region_data.values():
+        area_data.create_world_region(world)
+
+    impact_crater = Region("Impact Crater", world.player, world.multiworld)
+    world.multiworld.regions.append(impact_crater)
+
+    mission_complete = Region("Mission Complete", world.player, world.multiworld)
+    world.multiworld.regions.append(mission_complete)
+
+    assert world.starting_room_data and world.starting_room_data.name
+    starting_room = world.get_region(world.starting_room_data.name)
+    menu.connect(starting_room, "Starting Room")
+
+    if world.options.pre_scan_elevators:
+        # Pre-scanned, so no requirements.
+        can_access_elevator = lambda state: True

```

Comment:
The rule argument is optional. Not specifying a rule is functionally the same in most cases. Not specifying a rule is the tiniest bit better for memory usage and the tiniest bit faster if `add_rule` were to be used on the Entrance later on (`add_rule` does not need to chain the existing rule if there is no existing rule).
```suggestion
        can_access_elevator = None
```